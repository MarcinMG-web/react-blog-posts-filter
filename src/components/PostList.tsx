import { Post } from '../types/interface';

interface PostListProps {
  filteredPosts: Post[];
}

export default function PostList({ filteredPosts }: PostListProps) {
  return (
    <ul>
      {filteredPosts.map(({ id, title, body }) => (
        <li key={id}>
          <h2>{title}</h2>
          <p>{body}</p>
        </li>
      ))}
    </ul>
  );
}
