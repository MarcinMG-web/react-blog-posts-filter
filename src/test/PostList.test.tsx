import { screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { describe } from 'vitest';
import PostList from '../components/PostList';
import { render } from './test-utils';

describe('PostList component', () => {
  const mockPosts = [
    { id: 1, title: 'Post 1', body: 'Body 1', userId: 1 },
    { id: 2, title: 'Post 2', body: 'Body 2', userId: 1 },
    { id: 3, title: 'Post 3', body: 'Body 3', userId: 1 },
    { id: 4, title: 'Post 4', body: 'Body 4', userId: 1 },
  ];

  test('should render correctly', () => {
    const { asFragment } = render(<PostList filteredPosts={mockPosts} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('renders posts correctly', () => {
    render(<PostList filteredPosts={mockPosts} />);
    expect(screen.getByText('Post 1')).toBeInTheDocument();
    expect(screen.getByText('Post 2')).toBeInTheDocument();
    expect(screen.getByText('Post 3')).toBeInTheDocument();
  });

  test('paginates posts correctly', () => {
    render(<PostList filteredPosts={mockPosts} />);
    fireEvent.click(screen.getByText('Next'));

    expect(screen.queryByText('Post 1')).not.toBeInTheDocument();
    expect(screen.queryByText('Post 2')).not.toBeInTheDocument();
    expect(screen.queryByText('Post 3')).not.toBeInTheDocument();
    expect(screen.getByText('Post 4')).toBeInTheDocument();
  });

  test('disables previous button on first page', () => {
    render(<PostList filteredPosts={mockPosts} />);
    expect(screen.getByText('Previous')).toBeDisabled();
  });

  test('disables next button on last page', () => {
    render(<PostList filteredPosts={mockPosts} />);
    fireEvent.click(screen.getByText('Next'));
    expect(screen.getByText('Next')).toBeDisabled();
  });
});
