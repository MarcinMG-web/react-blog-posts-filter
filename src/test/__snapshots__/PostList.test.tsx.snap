// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PostList > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiStack-root css-qkdc1h-JoyStack-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-1f0d13i-JoyCard-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1squy4s-JoyTypography-root"
          data-first-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             Post 1
          </span>
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body-md css-khk740-JoyTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
            Body 1
          </span>
        </p>
        <span
          class="MuiTypography-root MuiTypography-body-xs css-1q0q8ha-JoyTypography-root"
          data-last-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-1f0d13i-JoyCard-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1squy4s-JoyTypography-root"
          data-first-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             Post 2
          </span>
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body-md css-khk740-JoyTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
            Body 2
          </span>
        </p>
        <span
          class="MuiTypography-root MuiTypography-body-xs css-1q0q8ha-JoyTypography-root"
          data-last-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiCard-root MuiCard-vertical MuiCard-variantOutlined MuiCard-colorNeutral MuiCard-sizeMd css-1f0d13i-JoyCard-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 css-1squy4s-JoyTypography-root"
          data-first-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             Post 3
          </span>
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body-md css-khk740-JoyTypography-root"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
            Body 3
          </span>
        </p>
        <span
          class="MuiTypography-root MuiTypography-body-xs css-1q0q8ha-JoyTypography-root"
          data-last-child=""
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-variantInline MuiSkeleton-levelInherit css-1h29701-JoySkeleton-root"
          >
             
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    class="MuiBox-root css-mfm10a"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-variantRectangular MuiSkeleton-levelInherit css-por1pp-JoySkeleton-root"
    >
      <button
        class="MuiButton-root Mui-disabled MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-1uzgewf-JoyButton-root"
        disabled=""
        type="button"
      >
        Previous
      </button>
    </span>
    <span
      class="MuiSkeleton-root MuiSkeleton-variantRectangular MuiSkeleton-levelInherit css-por1pp-JoySkeleton-root"
    >
      <button
        class="MuiButton-root MuiButton-variantSolid MuiButton-colorPrimary MuiButton-sizeMd css-1uzgewf-JoyButton-root"
        type="button"
      >
        Next
      </button>
    </span>
  </div>
</DocumentFragment>
`;
